### PR TITLE
Upgrade GeoServer, decrease max distance from 1 to 0

### PR DESCRIPTION
**Pull request by sbrunner**
Reintroduce pixel perfect generation of pdf to mapfish print
At first, I just wanted to set the tolerance parameter to 0,
then I encountered instability problems with GeoServeur, so I updated it,
the problems weren't solved, so I had to simplify a Shapefile.
I also encountered errors with the image layer, which I made more robust in the event of an image loading error.

### CODE REVIEW

File: examples/src/test/resources/examples/simple/expected_output/requestData-fontawesome.png
Comment (sebr72) [inline examples/src/test/resources/examples/simple/expected_output/requestData-fontawesome.png:1]: Check if easy to modify Countries to Fontawesome

File: core/src/main/java/org/mapfish/print/map/image/ImageLayer.java [hunk 1]
```diff
@@ -157,30 +169,47 @@ public double getImageBufferScaling() {
   public void prepareRender(
       final MapfishMapContext transformer,
       final MfClientHttpRequestFactory clientHttpRequestFactory) {
-    image = fetchImage(transformer, clientHttpRequestFactory);
+    try {
+      image = fetchImage(transformer, clientHttpRequestFactory);
+    } catch (Exception e) {
```
Comment (sebr72) [inline core/src/main/java/org/mapfish/print/map/image/ImageLayer.java:174]: please catch only the 2 exceptions from fetchImage and RuntimeException if required

File: core/src/main/java/org/mapfish/print/map/image/ImageLayer.java [hunk 2]
```diff
@@ -157,30 +169,47 @@ public double getImageBufferScaling() {
...
 
+    double widthImageBufferScaling = paintArea.getWidth() / transformer.getMapSize().getWidth();
+    double heightImageBufferScaling = paintArea.getHeight() / transformer.getMapSize().getHeight();
     imageBufferScaling =
-        ((paintArea.getWidth() / transformer.getMapSize().getWidth())
-                + (paintArea.getHeight() / transformer.getMapSize().getHeight()))
-            / 2;
+        Math.sqrt(
+            (Math.pow(widthImageBufferScaling, 2) + Math.pow(heightImageBufferScaling, 2)) / 2);
+    if (imageBufferScaling >= 2) {
```
Comment (sebr72) [inline core/src/main/java/org/mapfish/print/map/image/ImageLayer.java:196]: Please choose between removing or adding a config parameter to enable this feature, and include a test to show behaviour.

**Review by sebr72 — CHANGES_REQUESTED** [review]

File: examples/src/test/resources/examples/printwms_tyger_ny_EPSG_3857/expected_output/requestData-multi-layer-openlayers-2-compatible-style-wms.png
[the remark this answers is not in the document]
Comment (sbrunner) [reply examples/src/test/resources/examples/printwms_tyger_ny_EPSG_3857/expected_output/requestData-multi-layer-openlayers-2-compatible-style-wms.png:1]: It's the pixel of the `nurc:Img_Sample` layer :-)

**Review by sebr72 — APPROVED** [review]

**Pull request merged**